server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/symbols/SymbolAccumulatorService.java

```diff
@@ -328,10 +341,16 @@ public Optional<SyntaxError> registerFunctionNode(ProgramNode function) {
    * Search for a function reference
    *
    * @param name the name of the function
+   * @param programNode
    * @return the block reference or null if not found
    */
-  public FunctionInfo getFunctionReference(String name) {
-    return functions.get(name.toUpperCase());
+  public FunctionInfo getFunctionReference(String name, ProgramNode programNode) {
+    Boolean isDeclaredIntrinsic = Optional.ofNullable(programNode.getRepository().get(name.toUpperCase(Locale.ROOT))).orElse(false);
```

Comment:
Recursive?

Reply:
Yes. I am trying to figure out / create cases for nested programs. Will update these once I have test cases

Reply:
Yes.  we have a failing case below. Will add test case and update the code. Thanks!
```cobol
       IDENTIFICATION DIVISION.                             
       FUNCTION-ID. hex-of.                                  
       DATA DIVISION.                                       
       LINKAGE SECTION.                                     
       01  UNRELATED-STUFF.                                 
           05  NUM         PIC X(1234).                     
       PROCEDURE DIVISION RETURNING UNRELATED-STUFF.        
       END FUNCTION hex-of.                                  
       IDENTIFICATION DIVISION.                             
       PROGRAM-ID. PGM.                                     
       ENVIRONMENT DIVISION.                                
       CONFIGURATION SECTION.                               
       REPOSITORY.                                          
               FUNCTION hex-of intrinsic.                              
       DATA DIVISION.                                        
       WORKING-STORAGE SECTION.                             
       PROCEDURE DIVISION.
       IDENTIFICATION DIVISION.                                
       PROGRAM-ID. PGM1.                                            
       DATA DIVISION.                                       
       WORKING-STORAGE SECTION.                                       
       PROCEDURE DIVISION.                                   
             display hex-of.               
       END PROGRAM PGM1.                                                       
       END PROGRAM PGM.
```

Reply:
done